web/services/createTasksService.php

```diff
@@ -213,6 +215,11 @@
                     $string = "<return service='createTasks'><success>false</success><error id='4'>projectId is not valid</error></return>";
                     break;
                 }
+                //Support 0-hour tasks: reparse end time if initTime == 0 to the end so that order of parse doesn't cause error if end time added before init time by users
+                if (($endTimeParseOrig['hour']==0) && ($endTimeParseOrig['minute']==0) && ($initTime == 0)) {
+                    $endTime = $endTimeParseOrig['hour']*60 + $endTimeParseOrig['minute'];
```

Comment:
Isn't `$endTime` always 0 in this block, because `($endTimeParseOrig['hour']==0) && ($endTimeParseOrig['minute']==0)`?

Reply:
True, it will always be 0 in this case so we can hardcorde it directly

```suggestion
                    $endTime = 0;
```

Reply:
Good catch!